ports/analog/common-hal/busio/UART.c

```diff
@@ -0,0 +1,478 @@
+    bool sigint_enabled) {
+    int err, temp;
+
+    // Check for NULL Pointers && valid UART settings
+    assert(self);
+
+    // Assign UART ID based on pins
+    temp = pinsToUart(tx, rx);
+    if (temp == -1) {
+        // Error will be indicated by pinsToUart(tx, rx) function
+        return;
+    } else {
+        self->uart_id = temp;
+        self->uart_regs = MXC_UART_GET_UART(temp);
+    }
+    assert((self->uart_id >= 0) && (self->uart_id < NUM_UARTS));
+
+    // Check for size of ringbuffer, desire powers of 2
+    // At least use 1 byte if no size is given
+    assert((receiver_buffer_size & (receiver_buffer_size - 1)) == 0);
+    if (receiver_buffer_size == 0) {
+        receiver_buffer_size += 1;
+    }
+
+    // Indicate RS485 not implemented
+    if ((rs485_dir != NULL) || (rs485_invert)) {
+        mp_raise_NotImplementedError(MP_ERROR_TEXT("RS485"));
+    }
+
+    if ((rx != NULL) && (tx != NULL)) {
+        err = MXC_UART_Init(self->uart_regs, baudrate, MXC_UART_IBRO_CLK);
+        if (err != E_NO_ERROR) {
+            mp_raise_RuntimeError_varg(MP_ERROR_TEXT("%q init failed"), MP_QSTR_UART);
+        }
+
+        // attach & configure pins
+        self->tx_pin = tx;
+        self->rx_pin = rx;
+        common_hal_mcu_pin_claim(self->tx_pin);
+        common_hal_mcu_pin_claim(self->rx_pin);
+    } else {
+        mp_raise_NotImplementedError(MP_ERROR_TEXT("UART needs TX & RX"));
+    }
+
+    if ((cts) && (rts)) {
+        MXC_UART_SetFlowCtrl(self->uart_regs, MXC_UART_FLOW_EN, 8);
+        self->cts_pin = cts;
+        self->rts_pin = rts;
+        common_hal_mcu_pin_claim(self->cts_pin);
+        common_hal_mcu_pin_claim(self->rts_pin);
+    } else if (cts || rts) {
+        mp_raise_ValueError(MP_ERROR_TEXT("Both RX and TX required for flow control"));
+    }
+
+    // Set stop bits & data size
+    assert((stop == 1) || (stop == 2));
+    mp_arg_validate_int(bits, 8, MP_QSTR_bits);
+    MXC_UART_SetDataSize(self->uart_regs, bits);
+    MXC_UART_SetStopBits(self->uart_regs, stop);
+
+    // Set parity
+    MXC_UART_SetParity(self->uart_regs, convertParity(parity));
+
+    // attach UART parameters
+    self->stop_bits = stop; // must be 1 or 2
+    self->bits = bits;
+    self->parity = parity;
+    self->baudrate = baudrate;
+    self->error = E_NO_ERROR;
+
+    // Indicate to this module that the UART is active
+    uarts_active |= (1 << self->uart_id);
+
+    // Set the timeout to a default value
+    if (((timeout < 0.0) || (timeout > 100.0))) {
+        self->timeout = 1.0;
+    } else {
+        self->timeout = timeout;
+    }
+
+    // Initialize ringbuffer
+    if (receiver_buffer == NULL) {
+        self->ringbuf = m_malloc_without_collect(receiver_buffer_size);
+        if (!ringbuf_alloc(self->ringbuf, receiver_buffer_size)) {
+            m_malloc_fail(receiver_buffer_size);
+            mp_raise_RuntimeError(MP_ERROR_TEXT("ERR: Could not init ringbuffer\n"));
```

Comment:
Use an existing error message
```suggestion
            mp_raise_RuntimeError_var(MP_ERROR_TEXT("Failed to allocate %q buffer"), MP_QSTR_UART);
```